examples/plot_05_train_test_split.py

```diff
@@ -0,0 +1,180 @@
+"""
+.. _example_train_test_split:
+
+==========================
+Enhancing train-test split
+==========================
+
+This example illustrates the motivation and the use of skore's
+:func:`skore.train_test_split` to get assistance when developing your
+ML/DS projects.
+"""
+
+# %%
+# Creating and loading the skore project
+# ======================================
+#
+# We start by creating a temporary directory to store our project such that we can
+# easily clean it after executing this example. If you want to keep the project, you
+# have to skip this section.
+
+# %%
+import tempfile
+from pathlib import Path
+
+import skore
+
+temp_dir = tempfile.TemporaryDirectory(prefix="skore_example_")
+temp_dir_path = Path(temp_dir.name)
+
+my_project = skore.create("my_project.skore", working_dir=temp_dir_path)
+
+# %%
+# Train-test split in scikit-learn
+# ================================
+#
+# Scikit-learn holds a function for splitting the data into a training and a testing
+# sets: :func:`sklearn.model_selection.train_test_split`.
+# Its signature is the following:
+#
+# .. code-block:: python
+#
+#     sklearn.model_selection.train_test_split(*arrays, test_size=None, train_size=None, random_state=None, shuffle=True, stratify=None)
+#
+# where ``*arrays`` is a Python *args (it allows you to pass a varying number of
+# positional arguments) and the scikit-learn doc indicates that it is ``a sequence of
+# indexables with same length / shape[0]``.
+
+# %%
+# Let us construct a design matrix ``X`` and target ``y`` to illustrate our point:
+
+# %%
+import numpy as np
+
+X, y = np.arange(10).reshape((5, 2)), range(5)
+print(X, y)
+
+# %%
+# In scikit-learn, the most common usage is the following:
+
+# %%
+from sklearn.model_selection import train_test_split as sklearn_train_test_split
+
+X_train, X_test, y_train, y_test = sklearn_train_test_split(
+    X, y, test_size=0.2, random_state=0
+)
+print(X_train, y_train)
+print(X_test, y_test)
+
+# %%
+# Notice the shuffling that is done by default.
+
+# %%
+# In scikit-learn, the user can not explicitly set the design matrix ``X`` and
+# the target ``y``. The following:
+#
+# .. code-block:: python
+#
+#   X_train, X_test, y_train, y_test = sklearn_train_test_split(
+#       X=X, y=y, test_size=0.2, random_state=0)
+#
+# would return:
+#
+# .. code-block:: python
+#
+#   TypeError: got an unexpected keyword argument 'X'
+#
+# In general, in Python, positional arguments are useful to prevent typos such as:
```

Comment:
```suggestion
# In general, in Python, keyword arguments are useful to prevent typos. For example, in the following, ``X`` and ``y`` are reversed:
```